molten-cache/src/test/java/com/hotels/molten/cache/ReactiveMapCacheTest.java

```diff
@@ -65,4 +68,46 @@ public void shouldDelegatePutLazily() {
             .then(() -> verify(cache).put(KEY, VALUE))
             .verifyComplete();
     }
+
+    @Test
+    public void shouldPutViaOperator() {
```

Comment:
Please add test case which checks that in case of already cached value the original mono is not subscribed to.